storage/src/main/java/tech/pegasys/artemis/storage/MapDbDatabase.java

```diff
@@ -67,15 +73,58 @@
   private final ConcurrentNavigableMap<UnsignedLong, Set<Bytes32>> hotRootsBySlotCache =
       new ConcurrentSkipListMap<>();
 
+  private static void validateDatabaseVersion(File databaseVersionFile) {
+    try {
+      if (databaseVersionFile.exists()) {
+        Scanner contents = new Scanner(databaseVersionFile, UTF_8.name());
+        String ver = contents.nextLine();
+        if (StringUtils.isEmpty(ver) || !ver.equals(VERSION)) {
+          throw new DatabaseStorageException(
+              String.format(
+                  "The database version found (%s) does not meet the expected version(%s).\n"
+                      + "Aborting startup to prevent corruption of the database.\n",
+                  ver, VERSION));
+        }
+        STDOUT.log(
+            Level.INFO,
+            String.format(
+                "The existing database is version %s, from file: %s",
+                VERSION, databaseVersionFile.getAbsolutePath()));
+      } else {
+        STDOUT.log(
+            Level.INFO,
+            String.format(
+                "Recording database version %s to file: %s",
+                VERSION, databaseVersionFile.getAbsolutePath()));
+        FileWriter fileWriter = new FileWriter(databaseVersionFile.getPath(), UTF_8);
+        fileWriter.write(VERSION);
+        fileWriter.close();
```

Comment:
`Files.writeString` would be better here.  Otherwise you need to ensure the `close` is done in a finally block or use try-with-resources to ensure it gets closed even if an exception is thrown.